blogposts/announcing-sourcegraph-3.0.md

```diff
@@ -0,0 +1,243 @@
+
+Also, in case you missed it, [Sourcegraph is now open source](https://about.sourcegraph.com/blog/sourcegraph-is-now-open-source) (Apache-licensed).
+
+Want to hear about new features and releases as soon as they're available? Follow [@srcgraph](https://twitter.com/srcgraph).
+
+Ready to [install or upgrade a self-hosted Sourcegraph instance](#install-or-upgrade)? You can also [try these features on Sourcegraph.com for open-source code](https://sourcegraph.com).
+
+## Highlights
+
+### Search query cheatsheet
+
+We want anyone to be able to start using Sourcegraph for code search without reading a manual. That's why we made Sourcegraph search super fast and added one-click query suggestions (computed locally using heuristics) to refine your query.
+
+But when you're searching more than a couple times daily, learning Sourcegraph's search keywords (such as `repo:`) will make you more productive. Click the new <kbd>?</kbd> icon next to any search button to see a quick reference. See ["Search query syntax"](https://docs.sourcegraph.com/user/search/queries) for more info.
+
+[Try it out](https://sourcegraph.com/search?q=repo:%5Egithub.com/kubernetes/kubernetes%24+lang:go+-f:test+-f:vendor+controller) (click the <kbd>?</kbd> next to the search button):
+
+[![Search query cheatsheet](./announcing-sourcegraph-3.0/search_query_cheatsheet.png)](https://sourcegraph.com/search?q=repo:%5Egithub.com/kubernetes/kubernetes%24+lang:go+-f:test+-f:vendor+controller)
+
+### Code intelligence for all languages
+
+Two of the most common questions developers have while reading code are:
+
+- "Where is this thing defined?"
+- "What calls/uses this thing?"
+
+You can now answer these questions easily across all of your organization's code, with Sourcegraph's improved code intelligence (go-to-definition and find-references) for [N (TODO) languages](https://sourcegraph.com/extensions?query=category%3A%22Programming+languages%22).
+
+It works immediately for all code with zero configuration because it uses heuristics to provide fuzzy results by default. This is surprisingly helpful and fast, and it finds results in your other repositories, too.
+
+Want precise results? Many of the [N language extensions](https://sourcegraph.com/extensions?query=category%3A%22Programming+languages%22) also can provide precise results, by building and analyzing the code like a compiler, often by using a language server. See each extension's page (such as the [TypeScript/JavaScript extension](https://sourcegraph.com/extensions/sourcegraph/typescript)) for details.
+
+As long as you're aware of the limitations of the default (fuzzy) mode, you'll like it lot. For example, in the fuzzy mode, if multiple functions in your code are named `open`, the reference results for `open` will include false positives. (If it bothers you, toggle it off temporarily in the command palette or disable the language extension completely.)
```

Comment:
Once https://github.com/sourcegraph/sourcegraph/issues/2035 is complete, we won't have the concept of fuzzy vs precise anymore, right? It looks like the toggle for fuzzy mode enables/disables other language extensions https://github.com/sourcegraph/sourcegraph-basic-code-intel/blob/c25813e581f1e94b26eee1cff2e2b2615f91da7f/src/extension.ts#L41 but that won't be necessary once we have individual language extensions for all the popular languages.

Reply:
- The Fuzzy/Precise toggle button in the file header is only for 2.13 and earlier. 3.0 beta and newer do not show that button; instead they show the "Show/hide fuzzy results" in the panel header.
- Once https://github.com/sourcegraph/sourcegraph/issues/2035 is complete, we will still have these concepts, but they will not be exposed to the user. Fuzzy means "no lang server is configured", precise means "a lang server is configured".

Is that what you were thinking too?

Reply:
Hmm, I still can't think of a reason to display a "Show/hide fuzzy results" button.

Here's how I imagine the UX once basic-code-intel is replaced by individual language extensions:

- For languages that don't have language servers yet (everything except Go, TypeScript, and Python), the user will get code intel powered by ctags + search
- For languages that do:
  - If the user has not specified `serverUrl` yet, they will get code intel powered by ctags + search
  - If they have, they will get code intel powered by the language server

Does that match your expectation of the end result of this task of splitting basic-code-intel into individual extensions?

Reply:
Yeah, we are good. I didn’t intend to imply that the show/hide-fuzzy button would continue to be shown.

There are cases when you want fuzzy results while you’re waiting for precise results. Imagine browsing a big TS or Go repo and you just want to find references on a fairly unique name. We can handle that in 3.1 though.

Reply:
Updated this wording